index.js

```diff
@@ -58,12 +58,19 @@ function delay(ms){
 
 class Challenge {
 	constructor(options = {}){
+		this.module = "acme-dns-01-cloudflare";
 		this.options = options;
-		this.client = new cloudflare({
-			email: options.email,
-			key: options.key,
-			token: options.token
+		this.cfClient = new cloudflare({
+			email: options.client?.email || options.email,
+			key: options.client?.key || options.key,
+			token: options.client?.token || options.token
 		});
+		this.client = {
+
+			email: options.cfClient?.email || options.client?.email || options.email,
```

Comment:
Will `options.cfClient` ever be set? Perhaps this is meant to be `this.cfClient`? Also, what's the need for setting `this.client` at all? Is this something `greenlock` etc. relies on?

Reply:
It's the constructor, so it wouldn't be set in the `this` scope yet. However, its true that it'll probably wont ever be set in `options` (Same as the one below), that was just leftover code from when i didn't fully understand the issue

Reply:
As for why this.client is being set, thats the client greenlock seems to be storing in the options file; However, the cloudflare prototype seems to have problems being stringified to json, and it wasn't being stored. Changing it to a simple object without prototype solved it on my end